src/test/java/com/stripe/model/EventTest.java

```diff
@@ -39,5 +61,40 @@ public void testReserialize() throws Exception {
     assertEquals(reserializedEvent.getRequest().getIdempotencyKey(),
         event.getRequest().getIdempotencyKey());
     assertEquals(reserializedEvent.getType(), event.getType());
+
+    // because non-null `previousAttributes` is a map without equals being implemented
+    // comparing them directly returns inequality
+    assertNotEquals(reserializedEvent.getData(), event.getData());
+
+    // we can compare setting the map to null, and compare the untyped map separately
+    final EventData eventData = event.getData();
+    final Map<String, Object> previousAttributes = eventData.getPreviousAttributes();
+    eventData.setPreviousAttributes(null);
+
+    final EventData reserializedEventData = reserializedEvent.getData();
+    final Map<String, Object> reserializedPreviousAttributes =
+        reserializedEventData.getPreviousAttributes();
+    reserializedEventData.setPreviousAttributes(null);
+
+    // events without the map are equal
+    assertEquals(reserializedEvent.getData(), event.getData());
+    // comparing previous attribute map
+    assertEquals(
+        ApiResource.GSON.toJsonTree(reserializedPreviousAttributes),
+        ApiResource.GSON.toJsonTree(previousAttributes));
+
+    // comparing deserialized stripe object
+    StripeObject reserializedStripeObject = stubIntegrationApiVersion(
+        reserializedEventData, "2018-09-24").getObject();
+    StripeObject stripeObject = stubIntegrationApiVersion(
+        eventData, "2018-09-24").getObject();
+    assertEquals(reserializedStripeObject, stripeObject);
+  }
+
+  private EventData stubIntegrationApiVersion(EventData data,
```

Comment:
I didn't use the pattern of saving old Stripe version before the test and restore afterwards because they can be flakey when running parallel tests. Making a stub here instead.